### PR TITLE
Add guidance on moving from dbt-spark to dbt-databricks

### PR DESCRIPTION
**Pull request by dataders**
## Prerelease docs
If this change is related to functionality in a prerelease version of dbt (delete if not applicable):
- [ ] I've added versioning components, as described in ["Versioning Docs"](../contributing/versioningdocs.md)
- [ ] I've added a note to the prerelease version's [Migration Guide](../website/docs/docs/guides/migration-guide)

## Checklist
If you added new pages (delete if not applicable):
- [ ] The page has been added to `website/sidebars.js`
- [ ] The new page has a unique filename

If you removed existing pages (delete if not applicable):
- [ ] The page has been removed from `website/sidebars.js`
- [ ] An entry has been added to `_redirects`

### CODE REVIEW

File: website/sidebars.js [hunk 1]
```diff
@@ -74,6 +74,7 @@ const sidebarSettings = {
             slug: '/docs/guides/migration-guide',
           },
           items: [
+            "docs/guides/migration-guide/migrating-from-spark-to-databricks.md",
```
Comment (dataders) [inline website/sidebars.js]:
```suggestion
            "docs/guides/migration-guide/migrating-from-spark-to-databricks",
```

**Comment by dataders** [comment]
@ueshin, I'd love for you to take a look at this when you get a chance!

File: website/docs/docs/guides/migration-guide/migrating-from-spark-to-databricks.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+### dbt Core
+
+In dbt Core, migrating to the dbt-databricks adapter from dbt-spark requires that you:
+1. install the new adapter in your environment, and
+2. modify your target in your `~/.dbt/profiles.yml`
+
+This changes will be needed for all users of your project
+#### Example
+
+If you're using `dbt-spark` today to connect to a Databricks SQL Endpoint, the below examples show a good before and adter of how to authenticate. The cluster example is also effectively the same.
```
Comment (ueshin) [inline website/docs/docs/guides/migration-guide/migrating-from-spark-to-databricks.md]: nit: `adter` -> `after`?

File: website/docs/docs/guides/migration-guide/migrating-from-spark-to-databricks.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+The steps are virtually the same if you're using a Spark cluster.
+
+
+<File name='~/.dbt/profiles.yml'>
+
+```yaml
+your_profile_name:
+  target: my_databricks
```
Comment (ueshin) [inline website/docs/docs/guides/migration-guide/migrating-from-spark-to-databricks.md]: This should be `dev` or line 96 should be `my_databricks`?

File: website/docs/docs/guides/migration-guide/migrating-from-spark-to-databricks.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+In order to migrate to dbt-databricks, you must first be using `dbt-spark>1.0.0` as `dbt-databricks` is only supported from `>1.0`.
+
+## Why change to dbt-databricks?
+
+Current users of dbt with Databricks are about to have some new experiences with respect to authentication as well as features available via the delta format.
+
+### Authentication Simplification
+
+The authentication information has now simplified greatly. Previously you would have to provide a `cluster` or `endpoint` id which was hard to parse out of the http_path provided in the Databricks UI. Now it is simple enough  changes are authentication method is now the same regardless if you are using a Cluster or a SQL endpoint
```
Comment (ueshin) [inline website/docs/docs/guides/migration-guide/migrating-from-spark-to-databricks.md]: super nit: an extra space between `simple enough` and `changes are`.

**Review by ueshin — COMMENTED** [review]
Thanks for pinging me! I left some nits.
cc @superdupershant @allisonwang-db for another look.

File: website/docs/docs/guides/migration-guide/migrating-from-spark-to-databricks.md (hunk 3 above)
Comment (dataders) [reply website/docs/docs/guides/migration-guide/migrating-from-spark-to-databricks.md]: it can be whatever, but I agree `dev` is more conventional
```suggestion
  target: dev
```

File: website/docs/docs/guides/migration-guide/migrating-from-spark-to-databricks.md (hunk 4 above)
Comment (dataders) [inline website/docs/docs/guides/migration-guide/migrating-from-spark-to-databricks.md]:
```suggestion
The authentication information has now simplified greatly. Previously you would have to provide a `cluster` or `endpoint` id which was hard to parse out of the http_path provided in the Databricks UI. Now it is simple enough changes are authentication method is now the same regardless if you are using a Cluster or a SQL endpoint
```

Comment (dataders) [reply website/docs/docs/guides/migration-guide/migrating-from-spark-to-databricks.md]: totally valid!

File: website/docs/docs/guides/migration-guide/migrating-from-spark-to-databricks.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+<File name='~/.dbt/profiles.yml'>
+
+```yaml
+your_profile_name:
+  target: my_databricks
+  outputs:
+    dev:
+      type: spark
+      method: odbc
+      driver: '/opt/simba/spark/lib/64/libsparkodbc_sb64.so'
```
Comment (allisonwang-db) [inline website/docs/docs/guides/migration-guide/migrating-from-spark-to-databricks.md]: Do we need to specify this?

File: website/docs/docs/guides/migration-guide/migrating-from-spark-to-databricks.md (hunk 3 above)
Comment (ueshin) [reply website/docs/docs/guides/migration-guide/migrating-from-spark-to-databricks.md]: This change seems not applied yet.

File: website/docs/docs/guides/migration-guide/migrating-from-spark-to-databricks.md (hunk 5 above)
Comment (dataders) [reply website/docs/docs/guides/migration-guide/migrating-from-spark-to-databricks.md]: @allisonwang-db this is how a user who is currently using Databricks via dbt-spark would have their profile specified. Fortunately, it becomes much more simple for dbt-databricks

**Comment by dataders** [comment]
@ueshin,does a user who has been using dbt-spark on Databricks without Delta need to do anything special to ensure a smooth transition to dbt-databricks? do they need to add `delta: false` to their `dbt_project.yml`?

**Comment by ueshin** [comment]
@dataders 

> does a user who has been using dbt-spark on Databricks without Delta need to do anything special to ensure a smooth transition to dbt-databricks? 

That's a really good point!

When creating table without `using` clause, which happens when a user doesn't specify `file_format` with `dbt-spark`, Databricks uses `Delta` format by default, so it should be fine.

But it reminds me of the incremental strategy that is changed to `"merge"` by default. If a user didn't specify `incremental_strategy` before, they should specify `"append"` to be consistent.

**Comment by dataders** [comment]
> > does a user who has been using dbt-spark on Databricks without Delta need to do anything special to ensure a smooth transition to dbt-databricks?

> When creating table without `using` clause, which happens when a user doesn't specify `file_format` with `dbt-spark`, Databricks uses `Delta` format by default, so it should be fine.

what about when a user has `snapshot` or `incremental` materializations created initially with dbt-spark, can they still snapshot against the existing non-Delta tables and have the new versions be used in Delta? or will they have to migrate or re-create the existing non-Delta snapshot & incremental materializations in order to continue using them w/ dbt-databricks?

**Comment by ueshin** [comment]
Assuming the user connects to the same cluster:

- For `snapshot`, so far the generated queries are the same between `dbt-spark` and `dbt-databricks` except for the explicit `using` clause, and the file format  for snapshot tables must be `delta` or `hudi` that need to be explicitly specified. Thus, it should work without any changes.
- For `incremental`, the default strategy was changed as I mentioned, so if a user didn't explicitly specify `incremental_strategy`, they should specify `"append"` to be consistent.

cc @allisonwang-db Do you have any thoughts? or correct me if I'm wrong. I might be missing something.

File: website/docs/docs/guides/migration-guide/migrating-from-spark-to-databricks.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+---
+
+
+## Pre-requisites
+
+In order to migrate to dbt-databricks, your project must be compatible with `dbt 1.0` or greater as dbt-databricks is not supported pre `dbt 1.0`.
+
+## Why change to dbt-databricks?
+
+The Databricks team, in collaboration with dbt Labs, built on top of the foundation that the dbt Labs’ dbt-spark adapter provided, and they added some critical improvements. The dbt-databricks adapter offers an easier set up, as it only requires three inputs for authentication, and it also has more features available via the delta format.
```
Comment (bilalaslamseattle) [inline website/docs/docs/guides/migration-guide/migrating-from-spark-to-databricks.md]: Formatting nit: Delta should be capitalized.

File: website/docs/docs/guides/migration-guide/migrating-from-spark-to-databricks.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+
+Previously users had to provide a `cluster` or `endpoint` ID which was hard to parse out of the http_path provided in the Databricks UI. Now the dbt-databricks profile requires the same inputs regardless if you are using a Cluster or a SQL endpoint. All you need to provide is:
+- the hostname of the Databricks workspace
+- the HTTP path of the Databricks SQL endpoint or cluster
+- an appropriate credential
+
+
+### Better defaults
+
+With dbt-databricks, by default, dbt models will use the Delta format and expensive queries will be accelerated with the [photon engine](https://docs.databricks.com/runtime/photon.html). See [the caveats section of Databricks Profile documentation](https://docs.getdbt.com/reference/warehouse-profiles/databricks-profile#choosing-between-dbt-databricks-and-dbt-spark) for more information.
```
Comment (bilalaslamseattle) [inline website/docs/docs/guides/migration-guide/migrating-from-spark-to-databricks.md]: Nit: Photon should be capitalized

File: website/docs/docs/guides/migration-guide/migrating-from-spark-to-databricks.md [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+
+### Pure Python (Core only)
+
+A huge benefit to Core only users is that with the new dbt-databricks adapter, you no longer have to download an independent driver to interact with Databricks. The connection information is all embedded in a pure-python library, `databricks-sql-connector`.
```
Comment (bilalaslamseattle) [inline website/docs/docs/guides/migration-guide/migrating-from-spark-to-databricks.md]: Nit: Python should be capitalized

**Review by bilalaslamseattle — COMMENTED** [review]
Left some minor formatting/casing comments

File: website/docs/docs/guides/migration-guide/migrating-from-spark-to-databricks.md (hunk 8 above)
Comment (dataders) [reply website/docs/docs/guides/migration-guide/migrating-from-spark-to-databricks.md]:
```suggestion
A huge benefit to Core only users is that with the new dbt-databricks adapter, you no longer have to download an independent driver to interact with Databricks. The connection information is all embedded in a pure-Python library, `databricks-sql-connector`.
```

File: website/docs/docs/guides/migration-guide/migrating-from-spark-to-databricks.md (hunk 7 above)
Comment (dataders) [reply website/docs/docs/guides/migration-guide/migrating-from-spark-to-databricks.md]:
```suggestion
With dbt-databricks, by default, dbt models will use the Delta format and expensive queries will be accelerated with the [Photon engine](https://docs.databricks.com/runtime/photon.html). See [the caveats section of Databricks Profile documentation](https://docs.getdbt.com/reference/warehouse-profiles/databricks-profile#choosing-between-dbt-databricks-and-dbt-spark) for more information.
```

File: website/docs/docs/guides/migration-guide/migrating-from-spark-to-databricks.md (hunk 6 above)
Comment (dataders) [reply website/docs/docs/guides/migration-guide/migrating-from-spark-to-databricks.md]:
```suggestion
The Databricks team, in collaboration with dbt Labs, built on top of the foundation that the dbt Labs’ dbt-spark adapter provided, and they added some critical improvements. The dbt-databricks adapter offers an easier set up, as it only requires three inputs for authentication, and it also has more features available via the Delta file format.
```

**Comment by gwenwindflower** [comment]
@dataders we've got the Guides section ready to go now, so let me know if you're ready to launch this i can get you re-situated into the right place in the sidebar.js and we'll :shipit:

File: website/docs/guides/migration/tools/migrating-from-spark-to-databricks.md [hunk 9]
```diff
@@ -24,6 +24,8 @@ Previously users had to provide a `cluster` or `endpoint` ID which was hard to p
 
 With dbt-databricks, by default, dbt models will use the Delta format and expensive queries will be accelerated with the [Photon engine](https://docs.databricks.com/runtime/photon.html). See [the caveats section of Databricks Profile documentation](https://docs.getdbt.com/reference/warehouse-profiles/databricks-profile#choosing-between-dbt-databricks-and-dbt-spark) for more information.
 
+Additionally, dbt-databricks's default `incremental_strategy` is now `merge`. If you were using the default version of incremental materializations with dbt-spark, you were using "incremental_strategy=append", and if you'd like to continue doing so, you'll have to set this config specifically on your incremental models. Read more [about `incremental_strategy` in dbt](https://docs.getdbt.com/docs/building-a-dbt-project/building-models/configuring-incremental-models#about-incremental_strategy). If you had already specified `incremental_strategy=merge` on your incremental models, you do not need to change anything when moving to dbt-databricks, though you could remove the param as it is now the default.
```
Comment (dataders) [inline website/docs/guides/migration/tools/migrating-from-spark-to-databricks.md]: @asteichen how does this language sound to you?

**Comment by dataders** [comment]
> @dataders we've got the Guides section ready to go now, so let me know if you're ready to launch this i can get you re-situated into the right place in the sidebar.js and we'll :shipit:

I think I got the sidebar figured out! ready to 🚢

File: website/docs/guides/migration/tools/migrating-from-spark-to-databricks.md (hunk 9 above)
Comment (dataders) [reply website/docs/guides/migration/tools/migrating-from-spark-to-databricks.md]:
```suggestion
Additionally, dbt-databricks's default `incremental_strategy` is now `merge`. The default `incremental_strategy` with dbt-spark is `append`.
If you have been using the default `incremental_strategy=append` with dbt-spark, and would like to continue doing so, you'll have to set this config specifically on your incremental models. Read more [about `incremental_strategy` in dbt](https://docs.getdbt.com/docs/building-a-dbt-project/building-models/configuring-incremental-models#about-incremental_strategy).
If you already specified `incremental_strategy=merge` on your incremental models, you do not need to change anything when moving to dbt-databricks, though you could remove the param as it is now the default.
```

File: website/docs/guides/migration/tools/migrating-from-spark-to-databricks.md [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+
+Previously users had to provide a `cluster` or `endpoint` ID which was hard to parse out of the http_path provided in the Databricks UI. Now the dbt-databricks profile requires the same inputs regardless if you are using a Cluster or a SQL endpoint. All you need to provide is:
+- the hostname of the Databricks workspace
+- the HTTP path of the Databricks SQL endpoint or cluster
+- an appropriate credential
+
+
+### Better defaults
+
+With dbt-databricks, by default, dbt models will use the Delta format and expensive queries will be accelerated with the [Photon engine](https://docs.databricks.com/runtime/photon.html). See [the caveats section of Databricks Profile documentation](https://docs.getdbt.com/reference/warehouse-profiles/databricks-profile#choosing-between-dbt-databricks-and-dbt-spark) for more information.
```
Comment (dataders) [inline website/docs/guides/migration/tools/migrating-from-spark-to-databricks.md]:
```suggestion
With dbt-databricks, by default, dbt models will use the Delta format and expensive queries will be accelerated with the [Photon engine](https://docs.databricks.com/runtime/photon.html). See [the caveats section of Databricks Profile documentation](https://docs.getdbt.com/reference/warehouse-profiles/databricks-profile#choosing-between-dbt-databricks-and-dbt-spark) for more information. Any declared configurations of `file_format = 'delta'` are now redundant and can be removed.
```

**Review by dataders — COMMENTED** [review]
incorporating feedback from @asteichen

File: website/docs/guides/migration/tools/migrating-from-spark-to-databricks.md [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+
+#### Procedure
+
+An admin of the dbt Cloud project running on Databricks should take the following steps to migrate from using the generic Spark adapter to the Databricks-specfic adapter. This should not cause any downtime for production jobs, but we recommend that you schedule the connection change when there is not heavy IDE usage for your team to avoid disruption.
+
+1. Select Account Settings in the main navigation bar.
+2. On the Projects tab, scroll until you find the project you'd like to migrate to the new dbt-databricks adapter.
+3. Click the hyperlinked Connection for the project.
+4. Click the "Edit" button in the top right corner.
+5. Click "Delete Connection" at the bottom of the page, and then confirm ythat you'd like to delete the connection.
```
Comment (thorn14) [inline website/docs/guides/migration/tools/migrating-from-spark-to-databricks.md]: This will need to be updated for the new adaptor selection via radio button group.

File: website/docs/guides/migration/tools/migrating-from-spark-to-databricks.md [hunk 12]
```diff
@@ -0,0 +1,113 @@
+---
+title: "Migrating from dbt-spark to dbt-databricks"
+id: "migrating-from-spark-to-databricks"
+---
+
+
+## Pre-requisites
+
+In order to migrate to dbt-databricks, your project must be compatible with `dbt 1.0` or greater as dbt-databricks is not supported pre `dbt 1.0`.
```
Comment (Fleid) [inline website/docs/guides/migration/tools/migrating-from-spark-to-databricks.md]:
```suggestion
In order to migrate to dbt-databricks, your project must be compatible with `dbt 1.0` or greater as dbt-databricks is not supported pre `dbt 1.0`. [This guide](https://docs.getdbt.com/guides/migration/versions/upgrading-to-v1.0) will help you upgrade your project if necessary.
```

File: website/docs/guides/migration/tools/migrating-from-spark-to-databricks.md [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+
+## Why change to dbt-databricks?
+
+The Databricks team, in collaboration with dbt Labs, built on top of the foundation that the dbt Labs’ dbt-spark adapter provided, and they added some critical improvements. The dbt-databricks adapter offers an easier set up, as it only requires three inputs for authentication, and it also has more features available via the Delta file format.
+
+### Authentication Simplification
+
+Previously users had to provide a `cluster` or `endpoint` ID which was hard to parse out of the http_path provided in the Databricks UI. Now the dbt-databricks profile requires the same inputs regardless if you are using a Cluster or a SQL endpoint. All you need to provide is:
```
Comment (Fleid) [inline website/docs/guides/migration/tools/migrating-from-spark-to-databricks.md]:
```suggestion
Previously users had to provide a `cluster` or `endpoint` ID which was hard to parse out of the http_path provided in the Databricks UI. Now the [dbt-databricks profile](https://docs.getdbt.com/reference/warehouse-setups/databricks-setup) requires the same inputs regardless if you are using a Cluster or a SQL endpoint. All you need to provide is:
```

File: website/docs/guides/migration/tools/migrating-from-spark-to-databricks.md [hunk 14]
```diff
@@ -0,0 +1,113 @@
...
+- the hostname of the Databricks workspace
+- the HTTP path of the Databricks SQL endpoint or cluster
```
Comment (Fleid) [inline website/docs/guides/migration/tools/migrating-from-spark-to-databricks.md]:
```suggestion
- the HTTP path of the Databricks SQL warehouse or cluster
```

File: website/docs/guides/migration/tools/migrating-from-spark-to-databricks.md [hunk 15]
```diff
@@ -0,0 +1,113 @@
...
+
+When an admin changes the dbt Cloud's connection to use the dbt-databricks adapter instead of the dbt-spark adapter, your team will not lose their credentials. This makes migrating from dbt-spark to dbt-databricks straightforward as it only requires deleting the connection and re-adding the cluster/endpoint information. Both the admin and users of the project need not re-enter personal access tokens.
+
+#### Procedure
+
+An admin of the dbt Cloud project running on Databricks should take the following steps to migrate from using the generic Spark adapter to the Databricks-specfic adapter. This should not cause any downtime for production jobs, but we recommend that you schedule the connection change when there is not heavy IDE usage for your team to avoid disruption.
+
+1. Select Account Settings in the main navigation bar.
```
Comment (Fleid) [inline website/docs/guides/migration/tools/migrating-from-spark-to-databricks.md]:
```suggestion
1. Select **Account Settings** in the main navigation bar.
```

File: website/docs/guides/migration/tools/migrating-from-spark-to-databricks.md [hunk 16]
```diff
@@ -0,0 +1,113 @@
...
+3. Click the hyperlinked Connection for the project.
+4. Click the "Edit" button in the top right corner.
+5. Click "Delete Connection" at the bottom of the page, and then confirm ythat you'd like to delete the connection.
+
+    > :warning:, the only information that will be deleted is the cluster or endpoint host and id. Your credential will still be saved. See above note.
+5. After being brought back to the project overview page, click "Configure a connection" so that you can start the connection flow for the new dbt-databricks adapter.
+6. Select the Databricks adapter that is not listed as "to be depecated", and enter:
+    1. the `hostname`
+    2. `http_path`, and
+7. Click save.
```
Comment (Fleid) [inline website/docs/guides/migration/tools/migrating-from-spark-to-databricks.md]:
```suggestion
5. Select Databricks for the warehouse
6. Select Databricks (dbt-databricks) for the adapter and enter:
    1. the `hostname`
    2. the `http_path`
    3. optionally the catalog name
7. Click save.
```

**Review by Fleid — DISMISSED** [review]

**Comment by Fleid** [comment]
I couldn't resolve the conflict, so I restarted from a fresh copy of `current` and ported the changes there: https://github.com/dbt-labs/docs.getdbt.com/pull/2409
